### PR TITLE
feat: add support for nano APIs

### PR DESCRIPTION
**Pull request by pedroferreira1**
### Acceptance Criteria
- Add support for nano APIs: state, history, blueprint information.
- Add support for GET parameters with multi value.

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — COMMENTED** [review]
chore(test): We should include tests for the new apis.

issue(ci): The CI is failing due to isort issues
suggestion: Run `make fmt` because it runs isort and black. Or use `make check` since it also checks mypy issues

File: serverless.yml [hunk 1]
```diff
@@ -642,6 +642,69 @@ functions:
...
+          path: node_api/nc_state
+          method: get
+          cors: true
+          caching:
+            enabled: true
+            ttlInSeconds: 300
+            cacheKeyParameters:
+              - name: request.querystring.id
+              - name: request.querystring.fields
+              - name: request.querystring.balances
```
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply serverless.yml:676]: Luis helped me a bit and we found out that our caching plugin doesn't support multivalue caching (https://github.com/DianaIonita/serverless-api-gateway-caching?tab=readme-ov-file#limitations)

I created an issue to handle this: https://github.com/HathorNetwork/hathor-explorer-service/issues/318

For now, I've deleted these parameters from the cache.

File: serverless.yml [hunk 2]
```diff
@@ -654,6 +654,62 @@ functions:
...
+    package:
+      patterns:
+        - 'handlers/node_api.py'
+    layers:
+      - { Ref: PythonRequirementsLambdaLayer }
+    events:
+      - http:
+          path: node_api/nc_state
+          method: get
+          cors: true
```
Comment (luislhl) [inline serverless.yml:669]: Should we add a comment about why this one has no caching?

File: handlers/node_api.py [hunk 3]
```diff
@@ -311,3 +311,65 @@ def graphviz_dot_neighbors(
...
+    event: ApiGatewayEvent, _context: LambdaContext, node_api: Optional[NodeApi] = None
+) -> dict:
+    """Get blueprint information."""
+    node_api = node_api or NodeApi()
+    blueprint_id = event.query.get("blueprint_id")
+
+    if blueprint_id is None:
+        raise ApiError("invalid_parameters")
+
+    response = node_api.get_nc_blueprint_information(blueprint_id)
```
Comment (luislhl) [inline handlers/node_api.py:370]: Should we treat exceptions raised in this call?

We could have a `HathorCoreTimeout` raised here, for instance. 

It's also possible for a JSON-string to be returned by `HathorCoreClient` containing an `error` field, apparently: https://github.com/HathorNetwork/hathor-explorer-service/blob/57cc7294878bf185be743a4dac503a1d4025ccfa/gateways/clients/hathor_core_client.py#L158

Comment (pedroferreira1) [reply handlers/node_api.py:383]: It makes sense, I added an error handler for when the client class returns `None` or a dict with `error` on it, so we can return `500` to the client. https://github.com/HathorNetwork/hathor-explorer-service/pull/286/commits/2196b506114e2921eac4aacd8e7df13ff69b400f

I also created an issue to improve the error handling in general for the explorer service. https://github.com/HathorNetwork/hathor-explorer-service/issues/319

Regarding the  `HathorCoreTimeout`, it's already raised in the `HathorCoreClient` and handled by the `ApiGateway` decorator class.

File: serverless.yml (hunk 2 above)
Comment (pedroferreira1) [reply serverless.yml:672]: Done https://github.com/HathorNetwork/hathor-explorer-service/pull/286/commits/ee73a81c0239a77ed6010df6de7ae47619f61cf4

**Review by luislhl — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**